frontend/src/pages/Login.jsx

```diff
@@ -1,68 +1,229 @@
+                    type="text"
+                    required
+                    value={username}
+                    onChange={(e) => setUsername(e.target.value)}
+                />
+                <label>Username</label>
+                <i className="bx bxs-user"></i>
                 </div>
-                <div className="input-container">
-                    <label>Password:</label>
-                    <input
-                        type="password"
-                        value={password}
-                        onChange={(e) => setPassword(e.target.value)}
-                        className="input"
-                        placeholder="Enter your password"
-                    />
-                     {error.password && <p className="error">{error.password}</p>}
+                <div className="input-box animation" style={{ "--i": 19, "--j": 2 }}>
+                <input
+                    type="email"
+                    required
+                    value={email}
+                    onChange={(e) => setEmail(e.target.value)}
+                />
+                <label>Email</label>
+                <i className="bx bxs-envelope"></i>
                 </div>
-                <button type="submit" className="login-button">
-                    Login
+                <div className="input-box animation" style={{ "--i": 20, "--j": 3 }}>
+                <input
+                    type="password"
+                    required
+                    value={password}
+                    onChange={(e) => setPassword(e.target.value)}
+                />
+                <label>Password</label>
+                <i className="bx bxs-lock-alt"></i>
+                </div>
+                <button className="btn animation" style={{ "--i": 21, "--j": 4 }}>
+                Register
                 </button>
+
+                {signupError && (
+                <p className="error-message" style={{ fontSize: "14px", color: "red", margin: "15px", marginLeft: "20%" }}>
+                    Failed to save. Try again.
+                </p>
+                )}
+
+                <div className="logreg-link animation" style={{ "--i": 22, "--j": 5 }}>
+                <p>
+                    Already have an account?{" "}
+                    <a className="login-link" onClick={toggleActive} style = {{cursor: "pointer"}}>Login</a>
+                </p>
+                </div>
             </form>
-            <Link to="/signup" style={{color :'purple' }}>you don't an account!</Link>
             </div>
-        </div >
-    );
+
+            <div className="info-text register">
+            <h2 className="animation" style={{ "--i": 17, "--j": 0 }}>Welcome!</h2>
+            <p className="animation" style={{ "--i": 18, "--j": 1 }}>
+                Hope, You and your Family have a Great Day
+            </p>
+            </div>
+        </div>
+        </div>
+    </div>
+  );
 };
 
-export default Login;
+export default Login;
```

Comment:
this form is better than the previous one , it performs the required functionalities but it could be simplified further more  for refactoring purposes